plan/planbuilder.go

```diff
@@ -626,6 +626,44 @@ func (b *planBuilder) findDefaultValue(cols []*table.Column, name *ast.ColumnNam
 	return nil, ErrUnknownColumn.GenByArgs(name.Name.O, "field_list")
 }
 
+func (b *planBuilder) calculateGeneratedColumns(columns []*table.Column, onDups map[string]struct{}, mockPlan LogicalPlan) (igc InsertGeneratedColumns) {
+	for _, column := range columns {
+		if len(column.GeneratedExprString) == 0 {
+			continue
+		}
+		columnName := &ast.ColumnName{Name: column.Name}
+		columnName.SetText(column.Name.O)
+		expr, _, err := b.rewrite(column.GeneratedExpr, mockPlan, nil, true)
```

Comment:
It's unefficient to rewrite expression every time, because for the same table, the result of rewriting is also the same. Is there any way to store the expression directly ?